### PR TITLE
Storage feature/frontend

### PR DESCRIPTION
**Pull request by rzhang74**
*Issue #, if available:*

*Description of changes:*

S3 functionality.
From table cell list transcribe and comprehend data.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: app/lambda/create_health_care_professional/lambda_function.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def putItem(self, HealthCareProfessionalName):
 
     def handle(self, event, context):
         try:
-            name = event[DATASTORE_COLUMN_HEALTH_CARE_PROFESSSIONAL_NAME] if DATASTORE_COLUMN_HEALTH_CARE_PROFESSSIONAL_NAME in event else None
+            name = event["queryStringParameters"][DATASTORE_COLUMN_HEALTH_CARE_PROFESSSIONAL_NAME] if DATASTORE_COLUMN_HEALTH_CARE_PROFESSSIONAL_NAME in event["queryStringParameters"] else None
             id = self.putItem(name)
```
Comment (alexchirayath) [inline app/lambda/create_health_care_professional/lambda_function.py]: If there is no name, shouldnt this lambda  should return an error?

File: app/lambda/create_session/lambda_function.py [hunk 2]
```diff
@@ -60,6 +57,6 @@ def handle(self, event, context):
                 }
             }
         except Exception as e:
-            print(str(e))
+            print("Unexpected error: %s" % e)
```
Comment (alexchirayath) [inline app/lambda/create_session/lambda_function.py:60]: During error handline, we need to return the right error code in Lambda API. PLease refer https://aws.amazon.com/blogs/compute/error-handling-patterns-in-amazon-api-gateway-and-aws-lambda/

File: app/lambda/create_session/lambda_function.py [hunk 3]
```diff
@@ -20,38 +20,35 @@ def CurTimeToEpoch(self):
...
+            TranscribeS3Path = event["queryStringParameters"][DATASTORE_COLUMN_TRANSCRIBE_S3_PATH] if DATASTORE_COLUMN_TRANSCRIBE_S3_PATH in event["queryStringParameters"] else None
+            ComprehendS3Path = event["queryStringParameters"][DATASTORE_COLUMN_COMPREHEND_S3_PATH] if DATASTORE_COLUMN_COMPREHEND_S3_PATH in event["queryStringParameters"] else None
+
+            id = self.putItem(PatientId, HealthCareProfessionalId, SessionName, SessionId, TimeStampStart, TimeStampEnd, TranscribeS3Path, ComprehendS3Path)
             result = {DATASTORE_COLUMN_SESSION_ID : id}
             return {
             "isBase64Encoded": False,
             "statusCode": 200,
-            'body': json.dumps(result),
+            # 'body': json.dumps(result),
```
Comment (alexchirayath) [inline app/lambda/create_session/lambda_function.py]: If we dont need this, can remove the commented line?

File: app/lambda/get_transcription_comprehend/lambda_function.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                'Access-Control-Allow-Headers': '*',
+                'Access-Control-Allow-Methods': 'GET,POST,OPTIONS'
+                }
+            }
+        except Exception as e:
+            print(str(e))
+
+lambda_handler = GetTranscriptionComprehendLambda.get_handler()
+
+# def getKeyName(sessionId, category, fileType):
```
Comment (alexchirayath) [inline app/lambda/get_transcription_comprehend/lambda_function.py]: Please remove commented out code

File: app/lambda/get_transcription_comprehend/lambda_function.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        try:
+            print(event)
+            sessionId = event['queryStringParameters']['sessionId'] if 'sessionId' in event['queryStringParameters'] else None
+            print(sessionId)
+            bucket = 'mtastack-mtastackstorages3bucketc161f3b3-1tfncqzctldb1'
+            comprehend_key = self.getKeyName(sessionId,'comprehend','json')
+            transcribe_key = self.getKeyName(sessionId,'transcribe','txt')
+            print(comprehend_key)
+            print(transcribe_key)
+            client = boto3.client('s3', region_name='us-west-2')
```
Comment (alexchirayath) [inline app/lambda/get_transcription_comprehend/lambda_function.py]: You dont need to harcode the region in your Lambda.
AWS Lambda supports some default run time variable. Have a look at : https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html

You can access the variable via os.environ['AWS_REGION']

File: app/lambda/get_transcription_comprehend/lambda_function.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import json
+import sys
+
+sys.path.append("../")
+import boto3
+from lambda_base import LambdaBase
+from constant_variables import *
+
+class GetTranscriptionComprehendLambda(LambdaBase):
```
Comment (alexchirayath) [inline app/lambda/get_transcription_comprehend/lambda_function.py]: Can we rename this to "GetSessionDataLambda"?

Currently this lambda gets data related to only Comprehend and Transcription, but we may add in other data also. Keeping that in mind ,GetSessionDataLambda might be a better name?

File: app/lib/medical-transcription-analysis-stack.ts [hunk 7]
```diff
@@ -125,6 +125,17 @@ export class MedicalTranscriptionAnalysisStack extends cdk.Stack {
         });
 
         webAppS3Bucket.addToResourcePolicy(cloudfrontPolicyStatement);
+
+        // const cloudfrontStorageBucketPolicyStatement = new iam.PolicyStatement({
```
Comment (alexchirayath) [inline app/lib/medical-transcription-analysis-stack.ts]: Is there a reason we have this commented code or can we remove it?

**Review by alexchirayath — CHANGES_REQUESTED** [review]
See inline comments for more details

File: app/lambda/get_transcription_comprehend/lambda_function.py (hunk 5 above)
Comment (rzhang74) [reply app/lambda/get_transcription_comprehend/lambda_function.py]: Done.

File: app/lambda/get_transcription_comprehend/lambda_function.py (hunk 6 above)
Comment (rzhang74) [reply app/lambda/get_transcription_comprehend/lambda_function.py]: Done.

File: app/lib/medical-transcription-analysis-stack.ts (hunk 7 above)
Comment (rzhang74) [reply app/lib/medical-transcription-analysis-stack.ts]: Removed.

File: app/lambda/get_transcription_comprehend/lambda_function.py (hunk 4 above)
Comment (rzhang74) [reply app/lambda/get_transcription_comprehend/lambda_function.py]: Done.

File: app/lambda/create_health_care_professional/lambda_function.py (hunk 1 above)
Comment (rzhang74) [reply app/lambda/create_health_care_professional/lambda_function.py]: Handled in datastore layer.

File: app/lambda/create_session/lambda_function.py (hunk 3 above)
Comment (rzhang74) [reply app/lambda/create_session/lambda_function.py]: Removed.

File: app/lambda/create_health_care_professional/lambda_function.py (hunk 1 above)
Comment (rzhang74) [reply app/lambda/create_health_care_professional/lambda_function.py]: Adding the error handling layer in the lambda also.

File: app/lambda/create_session/lambda_function.py (hunk 2 above)
Comment (rzhang74) [reply app/lambda/create_session/lambda_function.py:60]: Added.

**Review by alexchirayath — APPROVED** [review]
LGTM
Thanks for making in the changes

**Pull request merged**